### PR TITLE
chore: setup TypeDoc for SDK documentation

### LINKED ISSUE #444: Enhance Documentation Using JSDoc/TsDoc Standards
The SDK codebase would benefit from consistent, high-quality documentation to improve maintainability, ease of use, and on-boarding for new developers. To achieve this, we propose adopting [JSDoc](https://jsdoc.app/) and/or [TSDoc](https://tsdoc.org/) standards throughout the project. This will provide a standardized way to document functions, classes, parameters, return types, and modules, making it easier to generate comprehensive, up-to-date documentation.

Please start with the [web3](https://github.com/alephium/alephium-web3/tree/master/packages/web3) package.

### PR DESCRIPTION
**Pull request by Mystic-Nayy**
Resolves #444 

- Added TypeDoc and typedoc-plugin-markdown for documentation generation
- Configured typedoc.json for project-specific settings
- Updated package.json with `build-docs` script for generating documentation

### CODE REVIEW

**Comment by Mystic-Nayy** [comment]
@h0ngcha0 I have removed them sir

File: packages/web3/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,10235 @@
+{
```
Comment (h0ngcha0) [inline packages/web3/package-lock.json:1]: This `package-lock.json` file is new, is it necessary?

Comment (Mystic-Nayy) [reply packages/web3/package-lock.json:1]: No it is not

**Comment by Mystic-Nayy** [comment]
@h0ngcha0 done

**Comment by pragmaxim** [comment]
Is it supposed to work from the project root or from packages/web3? Neither is working according to doc : 
root does not have any doc related script :
```
[master:!][/tmp/alephium-web3]
[07:38:29.584] $ pnpm run build-docs
 ERR_PNPM_NO_SCRIPT  Missing script: build-docs

Command "build-docs" not found. Did you mean "pnpm run build"?
```
packages/web3 fails : 
```
[master:!][/tmp/alephium-web3/packages/web3]
[07:36:28.546] $ pnpm install
../web3-react                            |  WARN  The field "resolutions" was found in /tmp/alephium-web3/packages/web3-react/package.json. This will not take effect. You should configure "resolutions" at the root of the workspace instead.
Scope: all 8 workspace projects
Done in 1.1s

[master:!][/tmp/alephium-web3/packages/web3]
[07:36:38.359] $ pnpm run build-docs

> @alephium/web3@1.9.0 build-docs /tmp/alephium-web3/packages/web3
> typedoc

[error] The plugin typedoc-plugin-markdown could not be loaded
[error] TypeError: Cannot read properties of undefined (reading 'addOutput')
    at load (file:///tmp/alephium-web3/node_modules/.pnpm/typedoc-plugin-markdown@4.3.1_typedoc@0.26.11_typescript@4.9.5_/node_modules/typedoc-plugin-markdown/dist/index.js:48:17)
```

**Pull request closed without merging**